app/models/user.rb

```diff
@@ -61,17 +61,20 @@ def self.find_for_oauth(auth, signed_in_resource = nil)
 
       # Create the user if it's a new registration
       if user.nil?
-        brand = Brand.create(name: auth.info.name)
+        brand = Brand.create!(name: auth.info.name)
 
         user = User.new(
           name:         auth.extra.raw_info.name,
           email:        email ? email : "#{TEMP_EMAIL_PREFIX}-#{auth.uid}-#{auth.provider}.com",
           password:     Devise.friendly_token[0, 20],
           brand:        brand,
           confirmed_at: Time.current, # Needed for devise_token_auth gem
+          email_subscription: true
         )
-
         user.save!
+        
+        # Subscribe user to Mailchimp newsletter
+        user.subscribe_to_newsletter
```

Comment:
This doesn't feel like this is the right place to perform this especially since these are going to be temp email addresses that are invalid. We should wait until we get a valid email address from the user, like when the select their subscription for the first time. That would be a better place to subscribe the user to our email list.